llama_index/response_synthesizers/factory.py

```diff
@@ -41,8 +39,8 @@ def get_response_synthesizer(
     refine_template = refine_template or DEFAULT_REFINE_PROMPT_SEL
     simple_template = simple_template or DEFAULT_SIMPLE_INPUT_PROMPT
 
-    service_context = service_context or ServiceContext.from_defaults(
-        callback_manager=callback_manager
+    service_context = (
+        service_context or ServiceContext.get_global() or ServiceContext.from_defaults()
```

Comment:
Now the callback manager isn't propagated here? Any reason for removing this?

Reply:
So the way I see it, the propagation of callbackmanager is somewhat confused at the moment. For instance, the tool use query engine just grabs it from the first tool it is initialized from.

We could discuss further and I could reduce controversial changes in this PR, but definitely think that the callback manager lifetime/propagation should be rethought a little bit alongside the async callbacks refactor.

Reply:
Yea, the propagation is definitely a little difficult right now, mostly because some classes don't use the service_context, so it has to be passed in by other means 😓 

I think the key thing is that if a user sets up a service context with a callback manager, that callback manager is expected to be used/propagate from the top level 🤔

Reply:
I see. Perhaps if the global service context has the callbackmanager (its not optional) then we can use the global one over the propagated callback.

Reply:
In my preliminary design, the callback manager is just going to be a global for the entire process, so this may not end up mattering much in the end

Reply:
That's fair! Maybe we should avoid any changes around it until that lands then?

(Also side note, can't wait to see the new callback manager 😍)

Reply:
Yep, I'll revert it and default to the global